File: LeetCode/Python/solutions/p0028.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        #
        # Solution v1: Brute Force
        #
        # Runtime: 130 ms @ (beats) 16.04%
        # Memory Usage: 14.6 MB @ (beats) 33.63%
        #
        # h_len = len(haystack)
        # n_len = len(needle)
        #
        # if n_len == 0:
        #     return 0
        # elif h_len == 0:
        #     return -1
        # elif h_len < n_len:
        #     return -1
        # elif h_len == n_len:
        #     return 0 if haystack == needle else -1
        # else:
        #     for i in range(h_len - n_len + 1):
        #         sub = haystack[i:(i + n_len)]
        #         if sub == needle:
        #             return i
        #
        # return -1

        try:
            h_len = len(haystack)
            n_len = len(needle)
            haystack.index(needle)
        except:
            return -1

File: LeetCode/Java/src/solutions/p0028_FindIndexofFirstOccurrenceinString/P0028.java
package solutions.p0028_FindIndexofFirstOccurrenceinString;


public class P0028 {
    public int strStr(String haystack, String needle) {
        /*
         * Solution v1: Two Pointers
         *
         * Runtime:  ms @ (beats) %
         * Memory Usage:  MB @ (beats) %
         */

        if (needle.length() > haystack.length()){
            return -1;
        }

        int j = 0;
        for (int i = 0; i < haystack.length(); i++) {
            int k = i;
            while (haystack.charAt(k) == needle.charAt(j)) {
                if (j == needle.length() - 1){
                    return i;
                }
                j++;
                k++;
                if (k == haystack.length()) {
                    break;
                }
            }
            j = 0;
        }
        return -1;


    }
}
